src/CoreWCF.CrankTests/CoreWCFPerfClient/Program.cs

```diff
@@ -0,0 +1,168 @@
+
+    class Program
+    {
+        private TestBinding _paramBinding = TestBinding.BasicHttp;
+        private TimeSpan _paramPerfMeasurementDuration = s_defaultPerfMeasurementDuration;
+        private string _paramServiceUrl = "";
+        private int _paramRequests = 0;
+        private bool _paramCloseAndOpen = false;
+
+        private readonly static TimeSpan s_defaultPerfMeasurementDuration = TimeSpan.FromSeconds(10);
+
+        static void Main(string[] args)
+        {
+
+            Program test = new Program();
+
+            if (test.ProcessRunOptions(args))
+            {
+                var startTime = DateTime.Now;
+                int requestTime = 0;
+                int request = 0;
+
+                Binding binding = null;
+                switch (test._paramBinding)
+                {
+                    case TestBinding.BasicHttp:
+                        binding = new BasicHttpBinding();
+                        break;
+                    case TestBinding.NetTcp:
+                        binding = new NetTcpBinding(SecurityMode.None);
+                        break;
+                    default:
+                        break;
+                }
+
+                TimeSpan measurementDurationPerTime = test._paramPerfMeasurementDuration ;
+
+                BenchmarksEventSource.Register("firstrequest", Operations.Max, Operations.Max, "First Request (ms)", "Time to first request in ms", "n0");
+                BenchmarksEventSource.Register("corewcfperf/requests", Operations.Max, Operations.Sum, "Requests (" + test._paramPerfMeasurementDuration.TotalMilliseconds + " ms)", "Total number of requests", "n0");
+                BenchmarksEventSource.Register("corewcfperf/rps/max", Operations.Max, Operations.Sum, "Requests/sec (max)", "Max requests per second", "n0");
+
+                startTime = DateTime.Now;
+                ChannelFactory<ISayHello> factory = new ChannelFactory<ISayHello>(binding, new EndpointAddress(test._paramServiceUrl));
+                factory.Open();
+                var client = factory.CreateChannel();
+                ((IClientChannel)client).Open();
+                var stopwatchFirstReq = new Stopwatch();
+                stopwatchFirstReq.Start();
+                var result = client.HelloAsync("hello world").Result;
+
+                BenchmarksEventSource.Measure("firstrequest", stopwatchFirstReq.ElapsedMilliseconds);
+
+                while (DateTime.Now <= startTime.Add(measurementDurationPerTime))
+                {
+                    var rtnResult = client.HelloAsync("hello world").Result;
+                    Console.WriteLine(rtnResult);
+                    request++;
+                    requestTime = request;
+                    if (requestTime >= test._paramRequests & test._paramCloseAndOpen)
+                    {
+                        try
+                        {
+                            ((IClientChannel)client).Close();
+
```

Comment:
If you switch to async, you will replace this with `await Task.Factory.FromAsync(((IClientChannel)client).BeginClose, ((IClientChannel)client).EndClose, null);` and the equivalent for the Open call below.

Reply:
Have updated